joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/ClassTests.scala

```diff
@@ -950,4 +951,37 @@ class ClassTests extends RubyCode2CpgFixture {
       }
     }
   }
+
+  "fixme" in {
+    val cpg = code("""
+        |class Api::V1::MobileController
+        |end
+        |""".stripMargin)
+
+    inside(cpg.typeDecl.name("MobileController").l) {
+      case mobileTypeDecl :: Nil =>
+        mobileTypeDecl.name shouldBe "MobileController"
+        mobileTypeDecl.fullName shouldBe "Test0.rb:<main>.Api.V1.MobileController"
+        mobileTypeDecl.astParentFullName shouldBe "Api.V1"
+        mobileTypeDecl.astParentType shouldBe NodeTypes.NAMESPACE_BLOCK
```

Comment:
Could you also test the resulting NAMESPACE_BLOCK? It should be `external=false`, which may mean we need to create it. If we need to create the namespace block, then doing it on the file is fine. This can also be linked, i.e.
```scala
NewNamespaceBlock()
  .filename("foo.rb")
  // add to diffGraph manually
  ```